packages/rum-core/src/performance-monitoring/metrics.js

```diff
@@ -93,6 +95,103 @@ function createLongTaskSpans(longtasks) {
   return spans
 }
 
+export function isPerfEntryTypeSupported(type) {
+  return (
+    PerformanceObserver &&
+    PerformanceObserver.supportedEntryTypes &&
+    PerformanceObserver.supportedEntryTypes.indexOf(type) > -1
+  )
+}
+
+/**
+ * First Input Delay, source: https://github.com/GoogleChromeLabs/first-input-delay/blob/master/src/first-input-delay.js
+ */
+
+export function polyfillFID() {
+  const eventTypes = [
+    'click',
+    'mousedown',
+    'keydown',
+    'touchstart',
+    'pointerdown'
+  ]
+  const listenerOpts = { passive: true, capture: true }
+  const startTimeStamp = new Date()
+
+  const pointerup = 'pointerup'
+  const pointercancel = 'pointercancel'
+
+  function onPointerDown(delay, event) {
+    function onPointerUp() {
+      recordFirstInputDelay(delay, event)
+      removePointerEventListeners()
+    }
+
+    function onPointerCancel() {
+      removePointerEventListeners()
+    }
+
+    function removePointerEventListeners() {
+      removeEventListener(pointerup, onPointerUp, listenerOpts)
+      removeEventListener(pointercancel, onPointerCancel, listenerOpts)
+    }
+
+    addEventListener(pointerup, onPointerUp, listenerOpts)
+    addEventListener(pointercancel, onPointerCancel, listenerOpts)
+  }
+
+  function recordFirstInputDelay(delay, event) {
+    if (!metrics.fid) {
+      let firstInputTimeStamp = new Date()
+
+      eachEventType(removeEventListener)
+
+      if (delay >= 0 && delay < firstInputTimeStamp - startTimeStamp) {
+        metrics.fid = { delay, event }
+      }
+    }
+  }
+
+  function onInput(event) {
+    if (event.cancelable) {
+      let isEpochTime = event.timeStamp > 1e12
+      let now = isEpochTime ? new Date() : performance.now()
+
+      let delay = now - event.timeStamp
+
+      if (event.type == 'pointerdown') {
+        onPointerDown(delay, event)
+      } else {
+        recordFirstInputDelay(delay, event)
+      }
+    }
+  }
+
+  function eachEventType(callback) {
+    eventTypes.forEach(function(eventType) {
+      callback(eventType, onInput, listenerOpts)
+    })
+  }
+
+  eachEventType(addEventListener)
+}
+
+export function createFirstInputDelaySpan(list) {
+  const fidEntries = list.getEntriesByType(FIRST_INPUT)
+  let firstInput = fidEntries[0]
+
+  if (firstInput && !metrics.wasHidden) {
+    let { name, startTime, processingStart } = firstInput
```

Comment:
We dont want to use the `target`? this would be useful in checking which element was responsible for the FID. 
https://wicg.github.io/event-timing/#dom-performanceeventtiming-target